streaming_providers/torbox/utils.py

```diff
@@ -68,9 +70,23 @@ def fetch_downloaded_info_hashes_from_torbox(user_data: UserData) -> list[str]:
 
 def select_file_id_from_torrent(torrent_info: dict[str, Any], filename: str) -> int:
     """Select the file id from the torrent info."""
-    for index, file in enumerate(torrent_info["files"]):
-        if filename in file["name"]:
-            return file["id"]
-    raise ProviderException(
-        "No matching file available for this torrent", "api_error.mp4"
-    )
+    files = torrent_info["files"]
+    exact_match = next((f for f in files if filename in f["name"]), None)
+    if exact_match:
+        return exact_match["id"]
+
+    # Fuzzy matching as a fallback
+    for file in files:
+        file["fuzzy_ratio"] = fuzz.ratio(filename, file["name"])
+    selected_file = max(files, key=lambda x: x["fuzzy_ratio"])
+
+    # If the fuzzy ratio is less than 50, then select the largest file
```

Comment:
Using the largest file is perfectly working for movies. However, if it's series, then we need to parse the filenames and match it with episode number. See the RD implementation https://github.com/mhdzumair/MediaFusion/blob/main/streaming_providers/realdebrid/utils.py#L109-L113

Reply:
Did the same for torbox and parse the file name for episode number similar to RD. It works for almost everything?

Reply:
Yes. Let's do the file matching with fuzzy if not then fallback to episode checking then finally fallback to max size.